### ISSUE TITLE
Add Affected Host Count to Job API Endpoint

### ISSUE THREAD

**Issue by munizal** [ansible/awx#11176]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY

When browsing jobs in the API (/api/v2/jobs/) there is a field called “limit”.  It lists out the hosts that the job was limited to touching based on the inventory selection.  Its listed out as a string, calling out the hosts separated by comma.
It would be very useful if there were a field in this view that called out the total host count based on the limit.

**Comment by MalfuncEddie** [comment]
I have a similar use case. (posted on the mailing list)

I am currently using the api to see what playbooks are scheduled .
From this api call I get an inventory and if it is defined a limit. I thought given these 2 parameters it would be easy to get a list of hosts (via the api) that shows the hosts that match the limit and the inventory.

This seems quite hard. The only way (in theory) I think I could get this to launch a playbook with  something like 

- name: show all the hosts matching the pattern, i.e. all but the group www
  when: ansible_limit is defined  
  debug:
    msg: "{{ item }}"
  with_inventory_hostnames:
    - {{ansible_limit }}


https://docs.ansible.com/ansible/latest/collections/ansible/builtin/inventory_hostnames_lookup.html

This would be the scenario
Query schedules
for every schedule get the inventory and limit
for every inventory and limit launch a template with inventory and limit that prints the hostnames using the with_inventory_hostnames module.

This is inefficient so I hope there is a better way to filter out the hosts of a template when the limit is set. 

So am I on the right (but painfully tedious) way or have I missed a  crucial part in the api documentation?

In short: How do I get a filtered inventory given an ansible limit parameter via the API
